lib/KCard/BaseCard.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+
+  <div
+    :class="$computedClass({ ':focus': $coreOutline })"
+  >
+    <li class="remove-list-style">
+
+      <h2
+        v-if="title !== null && headingLevel === 2"
+      >
+        <a :href="to">
+          <KTextTruncator :text="title" :maxLines="1" />
+        </a>
+      </h2>
+      <h3
```

Comment:
Hi @AllanOXDi - looks like some great foundational work here! One small suggestion is that you could consider using [`<component>`](https://vuejs.org/api/built-in-special-elements#component) here to manage this, rather than write out each condition separately, just to have this be a little bit more concise. 

I haven't actually tried to run this code, so it might need some adjustments, but as an example: 
```
<component
  :is="'h' + headingLevel"
  v-if="title !== null"
>
  <a :href="to">
    <KTextTruncator :text="title" :maxLines="1" />
  </a>
</component>
```
I think this will work with semantic HTML, and not just custom components that are created within the repository, although there might be some edge cases so it would be good to double check the documentation here and not just take my word for it 😂

Reply:
Thanks for the suggestion- I going to be trying it out and see